fix:  internal anchor syntax

davidcok opened the pull request:
<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

cpitclaudel commented on docs/DafnyRef/UserGuide.md:
```diff
@@ -1159,6 +1158,7 @@ older versions of Dafny.
 * `-noExterns` - ignore `extern` and `dllimport` attributes in the
   program.
 
+<a id="sec-function-syntax"/>
```
I think you need a newline after this to go back to markdown (otherwise the whole paragraph below is HTML)

davidcok commented:
I added the blank line (and fixed another bad line break). But despite that being the rule for many HTML blocks, the text after the HTML anchor did not render any differently -- so perhaps the blank line is not needed for anchor tags. The bank line does not hurt though.

davidcok commented on docs/DafnyRef/UserGuide.md:
```diff
@@ -1178,7 +1179,8 @@ older versions of Dafny.
     functions, so change those into the new syntax `ghost function` and
-    `ghost predicate`. Then, start using `-functionSyntax:4`. This will
+    `ghost predicate`. Then, start using \
+    `-functionSyntax:4`. This will
```
To keep markdown from line breaking between the `-` and `functionSyntax:4`
